packages/palette/src/elements/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,174 @@
+  position: relative;
+  width: 100%;
+`
+
+const Viewport = styled(Box)`
+  width: 100%;
+  overflow: hidden;
+`
+
+const Rail = styled(Box)`
+  display: flex;
+  height: 100%;
+  transition: transform 500ms;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`
+
+const ARROW_WIDTH = space(4)
+
+const Arrow = styled(Clickable)`
+  position: absolute;
+  top: 0;
+  height: 100%;
+  width: ${ARROW_WIDTH}px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+  user-select: none;
+  color: ${color("black30")};
+  transition: opacity 250ms, color 250ms;
+
+  > svg {
+    fill: currentColor;
+  }
+
+  &:hover,
+  &:focus {
+    outline: 0;
+    color: ${color("black100")};
+  }
+
+  &:disabled {
+    opacity: 0;
+    pointer-events: none;
+  }
+`
+
+const Next = styled(Arrow)`
+  right: 0;
+  transform: translateX(100%);
+`
+
+const Previous = styled(Arrow)`
+  left: 0;
+  transform: translateX(-100%);
+`
+
+const Cell = styled(Box)``
+
+export interface CarouselProps extends BoxProps {
+  children: JSX.Element | JSX.Element[]
+}
+
+/** Carousel */
+export const Carousel: React.FC<CarouselProps> = ({ children, ...rest }) => {
+  const viewportRef = useRef<HTMLDivElement | null>(null)
+  const startRef = useRef<HTMLButtonElement | null>(null)
+  const endRef = useRef<HTMLButtonElement | null>(null)
+
+  const cells = useMemo(
+    () =>
+      Children.toArray(children)
+        .filter(isValidElement)
+        .map(child => ({ child, ref: createRef<HTMLDivElement>() })),
+    []
+  )
+
+  const [pages, setPages] = useState([0])
+
+  const { index, handleNext, handlePrev, setCursor } = useCursor({
+    max: pages.length,
+  })
+
+  const init = () => {
+    const { current: viewport } = viewportRef
+    const els = cells.map(({ ref }) => ref.current)
+    const values = els.map(node => node.clientWidth)
+    setCursor(0)
+    setPages(paginate(viewport.clientWidth, values))
+  }
+
+  useEffect(() => {
+    init()
+    window.addEventListener("resize", init)
+    return () => {
+      window.removeEventListener("resize", init)
+    }
+  }, [])
+
+  const skipToEnd = () => {
+    if (!endRef.current) return
+    endRef.current.focus()
+    setCursor(pages.length - 1)
+  }
+
+  const skipToStart = () => {
+    if (!startRef.current) return
+    startRef.current.focus()
+    setCursor(0)
+  }
+
+  return (
+    <Container {...rest}>
+      <Skip ref={startRef} onClick={skipToEnd} mb={1}>
+        Skip to end of content
+      </Skip>
+
+      <nav>
+        <Previous onClick={handlePrev} disabled={index === 0}>
+          <ChevronIcon direction="left" width={30} height={30} />
+        </Previous>
+
+        <Next onClick={handleNext} disabled={index === pages.length - 1}>
+          <ChevronIcon direction="right" width={30} height={30} />
+        </Next>
+      </nav>
+
+      <Viewport ref={viewportRef as any}>
+        <Rail as="ul" style={{ transform: `translateX(-${pages[index]}px)` }}>
```

Comment:
The offset which is animated ☝️

Reply:
**suggestion (non-blocking)**: Is that animation duration governed by [L33](https://github.com/artsy/palette/pull/760/commits/8c1a7151dd2ab40213c85409b63612e344441039?file-filters%5B%5D=#diff-307ddca242ea3ceaabcc04d4d2c47b10R33) above? If so maybe pull that into a named constant?

Reply:
Comment in code